src/full/Agda/Syntax/Parser/Helpers.hs

```diff
@@ -124,43 +124,47 @@ mkValidName constructor' r s = do
         isValidId con (Id y) = do
           let x = rawNameToString y
               err = "in the name " ++ s ++ ", the part " ++ x ++ " is not valid"
-          case parse defaultParseFlags [0] (lexer return) x of
+          case parse defaultParseFlags [0] (lexer return) (x <> " ") of
```

Comment:
Why are we appending a trailing space after `x` before parsing it?

Reply:
Well the *idea* was to get
```
_ = A.(|.B
```
to tell you that the name part `(|` isn't valid because it's used in idiom brackets, instead of saying that the `|` part is used for `with` application - without the space this wouldn't lex as a `SymOpenIdiomBracket`, because the rule for `(|` only fires if there's a space after - but in reality the lexer doesn't allow `(` in those positions, so the error here will always be garbage. We can revert this